reframe/frontend/executors/__init__.py

```diff
@@ -162,15 +220,23 @@ def _notify_listeners(self, callback_name):
     def _safe_call(self, fn, *args, **kwargs):
         if fn.__name__ != 'poll':
             self._current_stage = fn.__name__
+            self._timestamp[f'{self._current_stage}_start'] = time.time()
 
         try:
             with logging.logging_context(self.check) as logger:
                 logger.debug('entering stage: %s' % self._current_stage)
-                return fn(*args, **kwargs)
+                ret = fn(*args, **kwargs)
+                self._timestamp[f'{self._current_stage}_finish'] = time.time()
+                self._timestamp['pipeline_end'] = time.time()
```

Comment:
Because I think it doesn't work correctly for the cases that we want to raise an error, and not just return the result.

Reply:
Strange, I'll check that.

Reply:
Can  you post an example? I can't see the reason.

Reply:
This, which is very similar to what we do here works as expected:

```python
import os
import sys


def foo():
    try:
        os.stat(sys.argv[1])
        return 'hello'
    except OSError as e:
        raise Exception('error') from e
    finally:
        print('finally')


if __name__ == '__main__':
    print(foo())
```

`finally` will be printed in any case.

Reply:
You are right, I was getting an error and I thought it was because finally doesn't work as expected. After all, the problem is that if I put the lines
```
self._timestamp[f'{self._current_stage}_finish'] = time.time()
self._timestamp['pipeline_end'] = time.time()
```
in the finally section we will get the timestamp after the `self.fail()`. Fail will print the final message `[     FAIL ] ... [compile: ...]` and this will cause a reframe error.